Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs

```diff
@@ -478,6 +484,190 @@ private void ScanAndProcessFiles(string folder, Dictionary<string, Action<string
+            foreach (CompilationPlatformInfo platform in AvailablePlatforms)
+            {
+                exporter.ConfigurationPlatforms.Add(new ConfigurationPlatformPair(UnityConfigurationType.InEditor, platform.Name));
+                exporter.ConfigurationPlatforms.Add(new ConfigurationPlatformPair(UnityConfigurationType.Player, platform.Name));
+            }
+
+            // Set the folders by scanning for "projects" with folder type in old projects
+            foreach (KeyValuePair<Guid, Project> folderPair in solutionFileInfo.Projects.Where(t => t.Value.TypeGuid == SolutionProject.FolderTypeGuid))
+            {
+                exporter.GetOrAddFolder(folderPair.Key, folderPair.Value.Name)
+                    .Children.AddRange(solutionFileInfo.ChildToParentNestedMappings.Where(t => t.Value == folderPair.Key).Select(t => t.Key));
+            }
+
+            Dictionary<AssetLocation, Tuple<Guid, string>> knownFolderMapping = GetKnownFolderMapping();
+
+            // Confiure known folders for generated projects
+            ProcessSetForKnownFolders(exporter, knownFolderMapping, CSProjects.Values, t => t.AssemblyDefinitionInfo.AssetLocation, t => t.Guid);
+
+            //TODO handle existing projects
+
+            exporter.AdditionalSections = solutionFileInfo.SolutionSections;
+
+            exporter.Write();
+        }
+
+        private SolutionProject GetDependencyProjectReference(string projectName, string relativePath)
+        {
+            SolutionProject toReturn = new SolutionProject(config.DependenciesProjectGuid, SolutionProject.CSharpProjectTypeGuid, projectName, relativePath, null, CSProjects.Values.Select(t => t.Guid));
+
+            void SetMapping(ConfigurationPlatformPair configPlatform) => toReturn.ConfigurationPlatformMapping.Add(configPlatform, new ProjectConfigurationPlatformMapping() { ConfigurationPlatform = configPlatform, EnabledForBuild = true });
+
+            SetMapping(new ConfigurationPlatformPair("Debug", "Any CPU"));
+            SetMapping(new ConfigurationPlatformPair("Release", "Any CPU"));
+
+            return toReturn;
+        }
+
+        private void ProcessSetForKnownFolders<T>(ISolutionExporter exporter, Dictionary<AssetLocation, Tuple<Guid, string>> knownFolderMapping, IEnumerable<T> set, Func<T, AssetLocation> getAssetLocation, Func<T, Guid> getGuidFunc)
+        {
+            foreach (T item in set)
+            {
+                if (knownFolderMapping.TryGetValue(getAssetLocation(item), out Tuple<Guid, string> guidNameTuple))
+                {
+                    exporter.GetOrAddFolder(guidNameTuple.Item1, guidNameTuple.Item2, isGenerated: true)
+                        .Children.Add(getGuidFunc(item));
+                }
+            }
+        }
+
+        private Dictionary<AssetLocation, Tuple<Guid, string>> GetKnownFolderMapping()
+        {
+            return new Dictionary<AssetLocation, Tuple<Guid, string>>()
```

Comment:
>Dictionary [](start = 23, length = 10)

nit: can this be a constant or static generated?

Reply:
Nope, cause of the config reference.

---
In reply to: [372602902](https://github.com/microsoft/MSBuildForUnity/pull/116#discussion_r372602902) [](ancestors = 372602902)